sway/desktop/output.c

```diff
@@ -341,64 +343,105 @@ static void render_view_surfaces(struct sway_view *view,
 	struct render_data data = {
 		.output = output,
 		.damage = damage,
+		.view = view,
 		.alpha = alpha,
 	};
 	output_view_for_each_surface(
 			view, &data.root_geo, render_surface_iterator, &data);
 }
 
+static void render_saved_view(struct sway_view *view,
+		struct sway_output *output, pixman_region32_t *damage, float alpha) {
+	struct wlr_output *wlr_output = output->wlr_output;
+
+	struct wlr_texture *texture = transaction_get_texture(view);
+	if (!texture) {
+		return;
+	}
+	struct wlr_box box = {
+		.x = view->swayc->current.view_x - output->swayc->current.swayc_x,
+		.y = view->swayc->current.view_y - output->swayc->current.swayc_y,
+		.width = view->swayc->current.view_width,
```

Comment:
Using the view width/height will render incorrectly if the surface size is different. This happens with unresizable windows for instance. We should crop surfaces instead.